### DISCUSSION TITLE
Must python2 still be supported?

### DISCUSSION THREAD

**Discussion by vermaete** [netdata/netdata#12464] [Q&A] [answered]
In the [documentation ](https://learn.netdata.cloud/docs/agent/collectors/python.d.plugin#how-to-write-a-new-module) to start writing a Python collector, there is mentioned the code should be valid for and Python2 and Python3.

But in the checklist to [submit a PR](https://learn.netdata.cloud/docs/agent/collectors/python.d.plugin#pull-request-checklist-for-python-plugins), this is not mentioned.

Is it still needed to have the Python code be working with Python2 that went end of life at [2020-01-01](https://www.python.org/downloads/)?

**Comment by ilyam8** [comment]
Hi, @vermaete. No, it is not required. I think we need to remove [the line  ](https://github.com/netdata/netdata/blob/cabf89dfebb5441e2249760fde14afdb3739c91c/collectors/python.d.plugin/README.md?plain=1#L80). Are you willing to do a PR?

**Comment by ilyam8** [comment, marked as answer]
Ok, I went forward and made a [PR](https://github.com/netdata/netdata/pull/12465). The line about mandatory compatibility with python2 is removed. Thanks for highlighting the problem, @vermaete.
